### PR TITLE
add QFN-24-1EP

### PR DESCRIPTION
**Pull request by dudelectric**
Push of 
**QFN-24-1EP_4x4mm_P0.5mm_EP2.15x2.15mm**  and **QFN-24-1EP_4x4mm_P0.5mm_EP2.15x2.15mm_ThermalVias** footprint 
for part **LED1642GWQTR**

Part Datasheet: https://www.st.com/resource/en/datasheet/led1642gw.pdf

### CODE REVIEW

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 1]
```diff
@@ -177,6 +177,70 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
   #suffix: '_Pad{pad_x:.2f}x{pad_y:.2f}mm_HandSolder'
   #include_suffix_in_3dpath: 'False'
 
+QFN-24-1EP_4x4mm_P0.5mm_EP2.15x2.15mm:
+  device_type: 'QFN'
+  #manufacturer: 'man'
+  #part_number: 'mpn'
+  size_source: 'http://ww1.microchip.com/downloads/en/PackagingSpec/00000049BQ.pdf#page=278'
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:184]: Wrong source

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 2]
```diff
@@ -177,6 +177,70 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
...
+  ipc_class: 'qfn' # 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  # custom_name_format:
+  body_size_x:
+    nominal: 4
+    tolerance: 0.15
+  body_size_y:
+    nominal: 4
+    tolerance: 0.15
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:193]: Please use min/nom/max notation as in the datasheet.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 3]
```diff
@@ -177,6 +177,70 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
...
+    minimum: 0.3
+    nominal: 0.4
+    maximum: 0.5
+
+  # size choosen for backwards compatibility
+  EP_size_x:
+    minimum: 2.0
+    nominal: 2.15
+    maximum: 2.25
+  EP_size_x_overwrite: 2.15
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:213]: Unnecessary, same as nominal.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 4]
```diff
@@ -177,6 +177,70 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
...
+  EP_size_y:
+    minimum: 2.0
+    nominal: 2.15
+    maximum: 2.25
+  EP_size_y_overwrite: 2.15
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:218]: See above

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 5]
```diff
@@ -177,6 +177,70 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
...
+  lead_width:
+    minimum: 0.18
+    nominal: 0.25
+    maximum: 0.3
+  lead_len:
+    minimum: 0.3
+    nominal: 0.4
+    maximum: 0.5
+
+  # size choosen for backwards compatibility
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:208]: What does that mean?

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 6]
```diff
@@ -177,6 +177,70 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
...
+
+  thermal_vias:
+    count: [3, 3]
+    drill: 0.2
+    # min_annular_ring: 0.15
+    paste_via_clearance: 0.1
+    # EP_num_paste_pads: [2, 2]
+    #paste_between_vias: 1
+    #paste_rings_outside: 1
+    EP_paste_coverage: 0.75
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:230]: Please leave default if there is no manufacturer recommendation.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 7]
```diff
@@ -177,6 +177,70 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
...
+    minimum: 2.0
+    nominal: 2.15
+    maximum: 2.25
+  EP_size_y_overwrite: 2.15
+  # EP_paste_coverage: 0.65
+  EP_num_paste_pads: [2, 2]
+
+  thermal_vias:
+    count: [3, 3]
+    drill: 0.2
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:224]: Please increase to 0.3 for cheaper manufacturability.

**Review by chschlue — CHANGES_REQUESTED** [review]

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml (hunk 7 above)
Comment (dudelectric) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml]: Thank you for the review, just committed changes

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml [hunk 8]
```diff
@@ -179,67 +179,51 @@ QFN-24-1EP_3x4mm_P0.4mm_EP1.65x2.65mm:
...
   thermal_vias:
     count: [3, 3]
-    drill: 0.2
-    # min_annular_ring: 0.15
+    drill: 0.3
     paste_via_clearance: 0.1
-    # EP_num_paste_pads: [2, 2]
-    #paste_between_vias: 1
-    #paste_rings_outside: 1
     EP_paste_coverage: 0.75
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/qfn-24.yaml:220]: Please remove.

**Review by chschlue — CHANGES_REQUESTED** [review]

**Review by chschlue — APPROVED** [review]
Thanks.

**Pull request merged**